File: Notepad/CS/BalloonSearch.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using DevComponents.DotNetBar;

namespace Notepad
{
	/// <summary>
	/// Summary description for BalloonSearch.
	/// </summary>
	public class BalloonSearch : DevComponents.DotNetBar.Balloon
	{
		private System.Windows.Forms.TextBox textBox1;
		private System.Windows.Forms.Button button1;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.PictureBox pictureBox1;
		private System.Windows.Forms.LinkLabel linkLabel1;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public BalloonSearch()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			System.Resources.ResourceManager resources = new System.Resources.ResourceManager(typeof(BalloonSearch));
			this.textBox1 = new System.Windows.Forms.TextBox();
			this.button1 = new System.Windows.Forms.Button();
			this.label1 = new System.Windows.Forms.Label();
			this.pictureBox1 = new System.Windows.Forms.PictureBox();
			this.linkLabel1 = new System.Windows.Forms.LinkLabel();
			this.SuspendLayout();
			// 
			// textBox1
			// 
			this.textBox1.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
			this.textBox1.Location = new System.Drawing.Point(72, 48);
			this.textBox1.Name = "textBox1";
			this.textBox1.Size = new System.Drawing.Size(160, 20);
			this.textBox1.TabIndex = 0;
			this.textBox1.Text = "";
			// 
			// button1
			// 
			this.button1.DialogResult = System.Windows.Forms.DialogResult.OK;
			this.button1.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
			this.button1.Location = new System.Drawing.Point(116, 72);
			this.button1.Name = "button1";
			this.button1.Size = new System.Drawing.Size(72, 24);
			this.button1.TabIndex = 1;
			this.button1.Text = "&Search";
			this.button1.Click += new System.EventHandler(this.button1_Click);
			// 
			// label1
			// 
			this.label1.BackColor = System.Drawing.Color.Transparent;
			this.label1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.label1.Location = new System.Drawing.Point(72, 32);
			this.label1.Name = "label1";
			this.label1.Size = new System.Drawing.Size(112, 16);
			this.label1.TabIndex = 2;
			this.label1.Text = "Find what:";
			// 
			// pictureBox1
			// 
			this.pictureBox1.BackColor = System.Drawing.Color.Transparent;
			this.pictureBox1.Image = ((System.Drawing.Bitmap)(resources.GetObject("pictureBox1.Image")));
			this.pictureBox1.Location = new System.Drawing.Point(8, 24);
			this.pictureBox1.Name = "pictureBox1";
			this.pictureBox1.Size = new System.Drawing.Size(64, 64);
			this.pictureBox1.TabIndex = 3;
			this.pictureBox1.TabStop = false;
			// 
			// linkLabel1
			// 
			this.linkLabel1.AutoSize = true;
			this.linkLabel1.BackColor = System.Drawing.Color.Transparent;
			this.linkLabel1.Font = new System.Drawing.Font("Verdana", 6.75F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.linkLabel1.Location = new System.Drawing.Point(8, 88);
			this.linkLabel1.Name = "linkLabel1";
			this.linkLabel1.Size = new System.Drawing.Size(61, 11);
			this.linkLabel1.TabIndex = 4;
			this.linkLabel1.TabStop = true;
			this.linkLabel1.Text = "Switch Style";
			this.linkLabel1.VisitedLinkColor = System.Drawing.Color.Blue;
			this.linkLabel1.LinkClicked += new System.Windows.Forms.LinkLabelLinkClickedEventHandler(this.linkLabel1_LinkClicked);
			// 
			// BalloonSearch
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(240, 104);
			this.Controls.AddRange(new System.Windows.Forms.Control[] {
																		  this.linkLabel1,
																		  this.pictureBox1,
																		  this.label1,
																		  this.button1,
																		  this.textBox1});
			this.Location = new System.Drawing.Point(0, 0);
			this.Name = "BalloonSearch";
			this.Activated += new System.EventHandler(this.BalloonSearch_Activated);
			this.Deactivate += new System.EventHandler(this.BalloonSearch_Deactivate);
			this.ResumeLayout(false);

		}
		#endregion

		private void BalloonSearch_Deactivate(object sender, System.EventArgs e)
		{
			this.Opacity=.75;
		}

		private void BalloonSearch_Activated(object sender, System.EventArgs e)
		{
			this.Opacity=1;
		}

		private void button1_Click(object sender, System.EventArgs e)
		{
			if(this.Owner!=null)
			{
				if(textBox1.Text=="")
					MessageBox.Show("Please enter search text.","Search",MessageBoxButtons.OK,MessageBoxIcon.Information);
				else
					((frmMain)this.Owner).SearchActiveDocument(textBox1.Text);
			}
		}

		private void linkLabel1_LinkClicked(object sender, System.Windows.Forms.LinkLabelLinkClickedEventArgs e)
		{
			if(this.BackColor==SystemColors.Info)
			{
				// Switch to Office 2003 style colors...
				ColorScheme scheme=new ColorScheme(eDotNetBarStyle.Office2003);
				this.BackColor=scheme.ItemCheckedBackground;
                this.BackColor2=scheme.ItemCheckedBackground2;
				this.BackColorGradientAngle=scheme.ItemCheckedBackgroundGradientAngle;
				this.Refresh();
			}
			else
			{
				this.BackColor=SystemColors.Info;
				this.BackColor2=Color.Empty;
				this.Refresh();
			}
		}
	}
}

File: RibbonPad/VB/BalloonSearch.vb
Imports System.Collections
Imports System.ComponentModel
Imports DevComponents.DotNetBar

Namespace Ribbonpad
	''' <summary>
	''' Summary description for BalloonSearch.
	''' </summary>
	Public Class BalloonSearch
		Inherits DevComponents.DotNetBar.Balloon
		Private textBox1 As TextBox
		Private WithEvents button1 As Button
		Private label1 As Label
		Private pictureBox1 As PictureBox
		Private WithEvents linkLabel1 As LinkLabel
		''' <summary>
		''' Required designer variable.
		''' </summary>
		Private components As System.ComponentModel.Container = Nothing

		Public Sub New()
			'
			' Required for Windows Form Designer support
			'
			InitializeComponent()

			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub

		''' <summary>
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If components IsNot Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Dim resources As New System.Resources.ResourceManager(GetType(BalloonSearch))
			Me.textBox1 = New TextBox()
			Me.button1 = New Button()
			Me.label1 = New Label()
			Me.pictureBox1 = New PictureBox()
			Me.linkLabel1 = New LinkLabel()
			Me.SuspendLayout()
			' 
			' textBox1
			' 
			Me.textBox1.BorderStyle = BorderStyle.FixedSingle
			Me.textBox1.Location = New Point(72, 48)
			Me.textBox1.Name = "textBox1"
			Me.textBox1.Size = New Size(160, 20)
			Me.textBox1.TabIndex = 0
			Me.textBox1.Text = ""
			' 
			' button1
			' 
			Me.button1.DialogResult = DialogResult.OK
			Me.button1.FlatStyle = FlatStyle.Flat
			Me.button1.Location = New Point(116, 72)
			Me.button1.Name = "button1"
			Me.button1.Size = New Size(72, 24)
			Me.button1.TabIndex = 1
			Me.button1.Text = "&Search"
'			Me.button1.Click += New System.EventHandler(Me.button1_Click)
			' 
			' label1
			' 
			Me.label1.BackColor = Color.Transparent
			Me.label1.Font = New Font("Microsoft Sans Serif", 8.25F, FontStyle.Bold, GraphicsUnit.Point, (CByte(0)))
			Me.label1.Location = New Point(72, 32)
			Me.label1.Name = "label1"
			Me.label1.Size = New Size(112, 16)
			Me.label1.TabIndex = 2
			Me.label1.Text = "Find what:"
			' 
			' pictureBox1
			' 
			Me.pictureBox1.BackColor = Color.Transparent
			Me.pictureBox1.Image = (CType(resources.GetObject("pictureBox1.Image"), Bitmap))
			Me.pictureBox1.Location = New Point(8, 24)
			Me.pictureBox1.Name = "pictureBox1"
			Me.pictureBox1.Size = New Size(64, 64)
			Me.pictureBox1.TabIndex = 3
			Me.pictureBox1.TabStop = False
			' 
			' linkLabel1
			' 
			Me.linkLabel1.AutoSize = True
			Me.linkLabel1.BackColor = Color.Transparent
			Me.linkLabel1.Font = New Font("Verdana", 6.75F, FontStyle.Regular, GraphicsUnit.Point, (CByte(0)))
			Me.linkLabel1.Location = New Point(8, 88)
			Me.linkLabel1.Name = "linkLabel1"
			Me.linkLabel1.Size = New Size(61, 11)
			Me.linkLabel1.TabIndex = 4
			Me.linkLabel1.TabStop = True
			Me.linkLabel1.Text = "Switch Style"
			Me.linkLabel1.VisitedLinkColor = Color.Blue
'			Me.linkLabel1.LinkClicked += New System.Windows.Forms.LinkLabelLinkClickedEventHandler(Me.linkLabel1_LinkClicked)
			' 
			' BalloonSearch
			' 
			Me.AutoScaleBaseSize = New Size(5, 13)
			Me.ClientSize = New Size(240, 104)
			Me.Controls.AddRange(New Control() { Me.linkLabel1, Me.pictureBox1, Me.label1, Me.button1, Me.textBox1})
			Me.Location = New Point(0, 0)
			Me.Name = "BalloonSearch"
'			Me.Activated += New System.EventHandler(Me.BalloonSearch_Activated)
'			Me.Deactivate += New System.EventHandler(Me.BalloonSearch_Deactivate)
			Me.ResumeLayout(False)

		End Sub
		#End Region

		Private Sub BalloonSearch_Deactivate(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Deactivate
			Me.Opacity=.75
		End Sub

		Private Sub BalloonSearch_Activated(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Activated
			Me.Opacity=1
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			If Me.Owner IsNot Nothing Then
				If textBox1.Text="" Then
					MessageBoxEx.Show("Please enter search text.","Search",MessageBoxButtons.OK,MessageBoxIcon.Information)
				Else
					CType(Me.Owner, frmMain).SearchActiveDocument(textBox1.Text)
				End If
			End If
		End Sub

		Private Sub linkLabel1_LinkClicked(ByVal sender As Object, ByVal e As LinkLabelLinkClickedEventArgs) Handles linkLabel1.LinkClicked
			If Me.BackColor=SystemColors.Info Then
				' Switch to Office 2003 style colors...
				Dim scheme As New ColorScheme(eDotNetBarStyle.Office2003)
				Me.BackColor=scheme.ItemCheckedBackground
				Me.BackColor2=scheme.ItemCheckedBackground2
				Me.BackColorGradientAngle=scheme.ItemCheckedBackgroundGradientAngle
				Me.Refresh()
			Else
				Me.BackColor=SystemColors.Info
				Me.BackColor2=Color.Empty
				Me.Refresh()
			End If
		End Sub
	End Class
End Namespace
